### PR TITLE
IMAGINGCLOUD-356

### PR DESCRIPTION
**Pull request by s-zubov**

### CODE REVIEW

File: asposeimagingcloud/api/imaging_api.py [hunk 1]
```diff
@@ -27,27 +27,83 @@
...
-    def __init__(self, app_key, app_sid, base_url=None, api_version=None):
-        self.api_client = ApiClient()
-        self.api_client.configuration.api_key['api_key'] = app_key
-        self.api_client.configuration.api_key['app_sid'] = app_sid
-        if base_url:
-            self.api_client.configuration.host = base_url
-        if api_version:
-            self.api_client.configuration.api_version = api_version
+
+    def __init__(self, app_key=None, app_sid=None, base_url=None,
```
Comment (s-zubov) [inline asposeimagingcloud/api/imaging_api.py:42]: Python does not support constructor overloads, so I've extended ImagingApi constructor with new optional parameters, making it compatible with previous versions. Also I've added two factory methods `ImagingApi.create_cloud` and `ImagingApi.create_on_premise` with more strict parameters to improve usability.

File: README.md [hunk 2]
```diff
@@ -90,6 +90,20 @@ Please, look at [Examples](EXAMPLES.md) document.
 
 # Asynchronous API
 Each API method has its' async version with *Async* postfix, using *multiprocessing.pool.ThreadPool*.
+Please, look at [Examples](EXAMPLES.md) document
+
+# Aspose Cloud-hosted service VS on-premise deployment (*experimental feature*)
```
Comment (MaxCrank) [inline README.md]: @s-zubov Please, move this section to correspond with other SDKs e.g. https://github.com/aspose-imaging-cloud/aspose-imaging-cloud-dotnet/tree/development

File: asposeimagingcloud/api/imaging_api.py (hunk 1 above)
Comment (MaxCrank) [reply asposeimagingcloud/api/imaging_api.py:42]: @s-zubov would you mind to remove these factory methods (at least for now)? Still, you are free to add a task with a proposition to unify all SDKs by adding such methods and assign it to me - I'll think about it a bit later and if approved, I'll re-assign it to you. Thanks.

File: README.md [hunk 3]
```diff
@@ -90,6 +90,20 @@ Please, look at [Examples](EXAMPLES.md) document.
...
+Starting from v19.7, you can choose either to use Aspose Cloud-hosted image processing service (the standard way) or the Docker image from Docker Hub deployed on-premise to serve the requests.
+The details about key differences and deployment process will be described on the dedicated Docker Hub page  as soon as it's released.
+
+To succeed with your on-premise service usage by the SDK, you need to:
+1. Use the new API class constructor with base URL required parameter, API version and debug mode optional parameters.
+```python
+imagingApi = ImagingApi("yourServiceUrl");
```
Comment (MaxCrank) [inline README.md:98]: @s-zubov this call doesn't seem to correspond with actual constructor which includes _on_premise_ parameter. Please, look at other SDKs which have a single constructor and follow the same approach. Thanks.

**Review by MaxCrank — CHANGES_REQUESTED** [review]

File: asposeimagingcloud/api/imaging_api.py (hunk 1 above)
Comment (s-zubov) [reply asposeimagingcloud/api/imaging_api.py:42]: @MaxCrank I've removed these methods, now there is only one constructor with multiple optional parameters. `on_premise` parameter is not set explicitly and is determined by input data (`app_sid`, `app_key` and `base_url`).

File: README.md (hunk 2 above)
Comment (s-zubov) [reply README.md]: @MaxCrank Thank you, I've moved this section.

File: README.md (hunk 3 above)
Comment (s-zubov) [reply README.md:98]: @MaxCrank I've refactored constructor, now `ImagingApi("yourServiceUrl");` is a correct call.

File: README.md (hunk 2 above)
Comment (MaxCrank) [reply README.md]: @s-zubov It still doesn't look like the one from .NET or other SDKs. Please, inspect the .NET SDK readme https://raw.githubusercontent.com/aspose-imaging-cloud/aspose-imaging-cloud-dotnet/development/README.md and take note on headers

> # Examples
> Please, look at [Examples](EXAMPLES.md) document.
> 
> ### Aspose Cloud-hosted service VS on-premise deployment
> Starting from v19.7, you can choose either to use Aspose Cloud-hosted image processing service (the standard way) or the Docker image from Docker Hub deployed on-premise to serve the requests.
> The details about key differences and deployment process is described on the dedicated Docker Hub page.
> 
> To succeed with your on-premise service usage by the SDK, you need to:
> 1. Use the new API class constructor with base URL required parameter, API version and debug mode optional parameters.
> ```csharp
> var imagingApi = new ImagingApi("yourServiceUrl");
> ```
> 2. Set *storage* or *storageName* parameters for each request where they're present (mandatory!).
> 
> # Tests
> Tests are intended for internal usage only.

**Review by MaxCrank — APPROVED** [review]

**Pull request merged**